recipes/newrelic/apm/java/linux.yml

```diff
@@ -184,434 +122,185 @@ install:
-          }
-
-          save_systemd_configuration() {
-            local SERVICE=$1
-            local CATALINA_HOME=$2
-
-            SERVICE_CONFIG_FILE=$(systemctl status $SERVICE | grep -oP '(?<=\().*(?=\; enabled)')
-            if [[ ! -z $SERVICE_CONFIG_FILE ]]; then
-
-              # 0 - check if tomcat appears to be yum-installed
-              TOMCAT_CONFIG_FROM_YUM=$(find / \( -path /proc -o -path /sys -o -path /tmp \) -prune -o -name tomcat.conf -print)
-              if [[ ! -z $TOMCAT_CONFIG_FROM_YUM ]]; then
-                YUM_CONFIG_ROOT=$(echo $TOMCAT_CONFIG_FROM_YUM | grep -oP '\K.*(?=/tomcat.conf)')
-                YUM_CUSTOM_CONFIG_DIR=$YUM_CONFIG_ROOT/conf.d
-                if [ -d $YUM_CUSTOM_CONFIG_DIR ]; then
-                  save_sysd_yum_configuration $SERVICE $YUM_CUSTOM_CONFIG_DIR
-                  return 0
+          sudo touch /etc/newrelic-java/.detected_pids
+          sudo chmod 755 /etc/newrelic-java/.detected_pids
+          sudo nri-lsi-java -list | tr "," "\n" | tr -d "[ " | tr -d " ]" >> /etc/newrelic-java/.detected_pids
+          sudo sort -o /etc/newrelic-java/.detected_pids /etc/newrelic-java/.detected_pids
+
+          # if differences in state are detected, swap pid file with detected and attempt to attach
+          new_ids=$(sudo comm -3 /etc/newrelic-java/.detected_pids /etc/newrelic-java/.pids | wc -l)
+          if [[ $new_ids -gt 0 ]]; then
+            sudo rm /etc/newrelic-java/.pids
+            sudo mv /etc/newrelic-java/.detected_pids /etc/newrelic-java/.pids
+            for JAVA_PID in $(sudo cat /etc/newrelic-java/.pids);
+            do
+              INTROSPECTION_DATA=$(sudo nri-lsi-java -introspect ${JAVA_PID})
+              isMatchingPattern=$(echo $INTROSPECTION_DATA | grep $pattern | grep -v grep | wc -l)
+              if [ "$isMatchingPattern" -gt 0 ]; then
+                if [ -f /etc/newrelic-java/.pids_in_docker ]; then
+                  # check if docker_pid exists, then check if pid is on a container
+                  if [[ "$JAVA_PID" == "$(cat /etc/newrelic-java/.pids_in_docker | grep $JAVA_PID | grep -v grep)" ]]; then
+                    CONTAINER_ID=$(sudo docker ps -q | xargs sudo docker inspect --format '{{"{{"}}.State.Pid{{"}}"}}, {{"{{"}}.ID{{"}}"}}' | sed  -n -e "s|$JAVA_PID, ||p")
+                    if [[ $CONTAINER_ID ]]; then
+                      if [[ "$renameDocker" == "N" ]]; then
+                        appName=$(sudo docker inspect --format '{{"{{"}}.Name{{"}}"}}' $CONTAINER_ID | tr -d '/')
+                      fi
+                      echo "Running Tomcat process detected in container '$appName' ($CONTAINER_ID)"
+                      OS_DETAILS=$(sudo docker exec $CONTAINER_ID sh -c "uname -a")
+                      if [[ $OS_DETAILS ]]; then
+                        if [[ $(echo $OS_DETAILS | grep -i 'ubuntu' | grep -v grep | wc -l) -gt 0 || $(echo $OS_DETAILS | grep -i 'debian' | grep -v grep | wc -l) -gt 0 ]]; then
```

Comment:
@mbruzina I'm not sure this will work for Debian version.
Should we use a similar implementation to what we've done above:: isYumInstalled=$(which yum 2>&1 || true)

Reply:
Yes, we should be consistent.  I'll update 👍